.ci/lib.sh

```diff
@@ -35,7 +35,9 @@ function clone_and_build() {
 
 	# Override branch if we are testing a PR.
 	[ -z "$pr_number" ] || branch="${target_branch}"
```

Comment:
I wonder if this condition should be moved too

Reply:
no, this is ok as we still want to checkout the other repos to the correct branch